### PR TITLE
Featu/auth config

### PR DESCRIPTION
**Pull request by nnixaa**
### Please read and mark the following check list before creating a pull request:

 - [] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md) guide.
 - [] I read and followed the [New Feature Checklist](https://github.com/akveo/nebular/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:

### CODE REVIEW

File: src/framework/auth/strategies/auth-strategy.ts [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 export abstract class NbAuthStrategy {
 
-  protected defaultConfig: any = {};
-  protected config: any = {};
+  protected defaultOptions: NbAuthStrategyOptions;
+  protected options: NbAuthStrategyOptions;
 
-  setConfig(config: any): void {
-    this.config = deepExtend({}, this.defaultConfig, config);
+  setOptions(options: any): void {
```
Comment (Tibing) [inline src/framework/auth/strategies/auth-strategy.ts]: As we have *NbAuthStrategyOptions* why do we use any?

File: src/framework/auth/strategies/default/default-strategy-options.ts [hunk 2]
```diff
@@ -18,30 +22,89 @@ export interface NbDefaultStrategyModule {
...
+    },
+    defaultErrors: ['Something went wrong, please try again.'],
+    defaultMessages: ['You have been successfully logged out.'],
   };
-  messages?: {
-    key?: string;
-    getter?: Function;
+  token = {
+    key: 'data.token',
+    getter: (module: string, res: HttpResponse<Object>, options: any) => getDeepFromObject(res.body,
```
Comment (Tibing) [inline src/framework/auth/strategies/default/default-strategy-options.ts]: Also, why do we use *options: **any*** in all getters?

File: src/framework/auth/strategies/default/default-strategy-options.ts [hunk 3]
```diff
@@ -18,30 +22,89 @@ export interface NbDefaultStrategyModule {
...
-  resetPass?: boolean | NbDefaultStrategyReset;
-  logout?: boolean | NbDefaultStrategyReset;
-  token?: {
-    key?: string;
-    getter?: Function;
+export class NbDefaultAuthStrategyOptions extends NbAuthStrategyOptions {
+  name = 'email';
+  baseEndpoint? = '/api/auth/';
+  login?: boolean | NbDefaultStrategyModule = {
+    alwaysFail: false,
```
Comment (Tibing) [inline src/framework/auth/strategies/default/default-strategy-options.ts]: Why do we need *alwaysFail* in default properties? Maybe we have to move it in some specific Options for tests? I don't think it'll be very useful in production.

File: src/framework/auth/strategies/default/default-strategy-options.ts (hunk 2 above)
Comment (nnixaa) [reply src/framework/auth/strategies/default/default-strategy-options.ts]: as it's getting to long :)

File: src/framework/auth/strategies/default/default-strategy-options.ts (hunk 3 above)
Comment (nnixaa) [reply src/framework/auth/strategies/default/default-strategy-options.ts]: this is the default strategy, ex-email-pass strategy

**Pull request merged**